### PR TITLE
update dependencies

### PR DESCRIPTION
**Pull request by ritchie46**
This PR updates dependencies to the latest release. For `packed_simd` this was also necessary to be able to run on the latest nightly.

### CODE REVIEW

**Comment by Dandandan** [comment]
These versions should already be taken care of as they are SemVer compatible to the versions in `Cargo.toml` according to the SemVer rules of cargo (`= `means semver compliant in crates). I also checked it by running  cargo update` and looking into the Cargo.lock file (which is 0.3.5 and 1.5.4).

File: arrow/Cargo.toml [hunk 1]
```diff
@@ -43,9 +43,9 @@ indexmap = "1.6"
 rand = "0.7"
 csv = "1.1"
 num = "0.4"
-regex = "1.3"
+regex = "1.5"
```
Comment (alamb) [inline arrow/Cargo.toml:46]: I didn't think this would do anything (as cargo should be able to update `1.3` --> `1.5` automatically

**Review by alamb — APPROVED** [review]
LGTM -- thanks @ritchie46

File: arrow/Cargo.toml (hunk 1 above)
Comment (jorgecarleitao) [reply arrow/Cargo.toml:46]: don't we need to use `^1.0` for it to use the latest?

Comment (Dandandan) [reply arrow/Cargo.toml:46]: Not needed, caret does the same as without:

https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio

Comment (ritchie46) [reply arrow/Cargo.toml:46]: The implicit caret is indeed a bit confusing. But I will close this one, as it implicitly is solved. :)

**Pull request closed without merging**